### DISCUSSION TITLE
Uncaught (in promise) Error: [🍍]: getActivePinia was called with no active Pinia.

### DISCUSSION THREAD

**Discussion by sdoerger** [vuejs/pinia#886] [Help and Questions]
Hi there, again great library. Replaced Vuex with it in my project.
However, I created myself a helper class, because I created them (like 20) always in the same way.
Then I thought, that class would have been great, when I started out with Vue and that would be a great first npm package project for me too.

To test it, I used `npm link` in my project, which worked great, but if I install it with `npm install` I get that error ... why?

As mentioned, first time that I created anything like that.
Any chance you can have a look at my [repo](https://github.com/sdoerger/retonio)?
It is just a single index file (because import form "./src/..." did not work)

Sorry if I waste time for very likely not properly at all, rushing in building a npm package. But if it is something kinda tiny or you can point me into the right direction, that would be really great.

Thanks in advance

**Comment by posva** [comment]
This is because of how linking keeps relative dependencies, e.g. your linked library will keep its relative dependency to pinia, vue, etc while your app using the linked library will use its own relative dependency of pinia, vue, etc.

And that will create problems when it comes to `Symbol`s, ending up in the app and library using different one and not sharing the same reactivity system.

Personally, I use https://github.com/wclr/yalc to deal with this.

**Reply by sdoerger** [reply]
Thank you so much for this god speed answer. I assumd something like that, but had no idea where to start searching for.
I gonna try it out.
... isn't there any lib, that somebody, who installs my app, can use it as is ... that my app tells the parent app "hey, you suppose to have pinia installed and if so my app will know where it is and use the same instance"?

Thank you very much.

**Reply by sdoerger** [reply]
Sorry to bother again.
I just went into my repo and did `yalc publish` and in the repo where I use it, I did `yalc add retoniojs@0.0.6`. That throw the same error as before.
Then I went into parent/node_modules/retoniojs and did there `yalc publish` and `yalc push` and again in the root again `yalc add retoniojs@0.0.6` ... is that wrong or something missing?

Thank you much.

**Reply by posva** [reply]
aah, yeah, you also need to call `app.use(pinia)` in the app using your lib. [This answer could also be useful](https://github.com/posva/pinia/discussions/856#discussioncomment-1727384)

**Reply by sdoerger** [reply]
As deeper I get, as more confused I am.
In my parent app (lets call that, the Vue app) I call `const app = createApp(App); app.use(pinia).... ` in the main.ts file. Do you mean besides that again ...?
I tried some things like on the answer link, but it does not work.
Maybe not, in the other question, it was mentioned that there is no main.ts where it is init.

[Here in index.js](https://github.com/sdoerger/retonio/blob/main/index.js) I tried in line 60 to call createPinia() but with no success either. Still the same error :/

**Reply by posva** [reply]
Maybe you are falling in one of those cases: https://pinia.esm.dev/core-concepts/outside-component-usage.html#using-a-store-outside-of-a-component.
You shouldn't be creating a pinia inside of your lib, **only create one pinia** inside of your app

**Reply by sdoerger** [reply]
Yes, seemed to be strange for me too, but when desperation come, logic goes ...
Anyway, I just got a different idea: I just pass "defineStore" from Pinia in the parent Vue app, via a config var, to my function from my lib, which creates a new Pinia store from the "root" Pinia store of the .

So a few lines more, but it works :D
I am really happy about hat.
Thanks for taking you time, explaining and pointing into directions. Really appreciate it.
